include/qt_atomics.h

```diff
@@ -149,18 +149,20 @@ extern pthread_mutexattr_t _fastlock_attr;
 # define QTHREAD_TRYLOCK_INIT(x)     { (x).u = 0; }
 # define QTHREAD_TRYLOCK_INIT_PTR(x) { (x)->u = 0; }
 # define QTHREAD_TRYLOCK_LOCK(x)     { uint32_t val = qthread_incr(&(x)->s.users, 1); \
-                                       while (val != (x)->s.ticket) SPINLOCK_BODY(); \
-                                          THREAD_FENCE_MEM_ACQUIRE; /* spin waiting for my turn */ }
+                                       while (val != atomic_load_explicit((_Atomic uint32_t *)&(x)->s.ticket, memory_order_acquire)) SPINLOCK_BODY();\
+                                       THREAD_FENCE_MEM_ACQUIRE;}
 # define QTHREAD_TRYLOCK_UNLOCK(x)   do { COMPILER_FENCE; \
-					                      THREAD_FENCE_MEM_RELEASE; \
+					  THREAD_FENCE_MEM_RELEASE; \
                                           qthread_incr(&(x)->s.ticket, 1); /* allow next guy's turn */ \
                                         } while (0)
 # define QTHREAD_TRYLOCK_DESTROY(x)
 # define QTHREAD_TRYLOCK_DESTROY_PTR(x)
 
 static inline int QTHREAD_TRYLOCK_TRY(qt_spin_trylock_t *x)
 {
-    qt_spin_trylock_t newcmp, cmp = *(x);
+    qt_spin_trylock_t newcmp, cmp;
+    uint64_t tmp = atomic_load_explicit((_Atomic uint64_t*)x, memory_order_acquire);
```

Comment:
I like the change. Likely this code used to work prior to this change as well due to the fence in the loop in L175.

Reply:
Interesting point. I'm not sure the thread fence at 175 even matters because `qthread_cas` already has `acq_rel` consistency since it's implemented using the old gcc sync interface for atomics.

That aside, the main reasoning for this specific change is that we're already using an atomic access at this location in 174 and mixing atomic and non-atomic accesses is undefined behavior. I don't remember now why I added the acquire consistency here though. I'll go back and check.

Reply:
Yes - I think we should be able to remove `THREAD_FENCE_MEM_ACQUIRE`. Let's see if we break something :)